ls1intum/Artemis
# `Discussion`: Replace answer popup with in-place message

### Checklist
#### General


#### Client

### Motivation and Context


While creating or editing answers to posts, users are provided with a pop-up modal where they can enter the answer body. Although the feature is functioning at the moment, users are losing their attention from the post itself when submitting an answer.

### Description

To let the user still see the post they are answering, we replace the post answer pop-up with container rendering. This way, while editing or creating a new answer, users are still able to see and interact with the post itself

Note that we leave post creation and editing as it is since there are multiple columns for posts versus answer posts. Also, each post consists of its own topic, so users do not need to see other posts while altering a post anyway.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with postings enabled 

1. Log in to Artemis
2. Navigate to the Discussions page of a course
3. Reply to a post and edit it
4. Make sure that the markdown editor is rendered without the pop-up.
6. Make sure that post creation and editing still works as before via the pop-up.
7. Go to PAGE_SECTION view for Discussions. (Navigate to a specific exercise or lecture)
8. Repeat the steps above on the discussion tab to the right.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| answer-post.component.ts | 100% | 100% |
| posting-create-edit-modal.directive.ts | 80% | 100% |
| answer-post-create-edit-modal.component.ts | 77.41% | 93.54% | 
| post-footer.component.ts | 88% | 100% |
| answer-post-header.component.ts | 81.81% | 100% |


### Screenshots


![image](https://user-images.githubusercontent.com/67344158/170456027-4766dccd-bd9b-4999-905d-40e6e1455333.png)
![image](https://user-images.githubusercontent.com/67344158/170456111-24d0212c-9351-4059-ada7-b0fee0a22c38.png)
![image](https://user-images.githubusercontent.com/67344158/170456228-03062627-81a2-4286-a02d-3dc9bdcc32a9.png)
https://user-images.githubusercontent.com/67344158/170456887-7f5a6874-639a-44e1-940d-c1641e70293d.mp4
